docs/customization.rst

```diff
@@ -188,3 +188,19 @@ With this simple handler only ``Done!`` will be displayed at the end of the test
 - Store logs in a custom format
 - Change the output visual style
 - Display additional information in the output
+
+``add_case``
+~~~~~~~~~~~~
+
+For each ``add_case`` hook and for each endpoint, we create an additional, duplicate test case. We pass the Case object from the duplicate test to the ``add_case`` hook.
+The user may change the Case object (and therefore the request data) before the request is sent to the server. The ``add_case`` allows the user to target specific
+behavior in the API by changing specific details of the duplicate request.
+
+.. code:: python
+
+    def add_case(context: HookContext, case: Case) -> Case:
+        case.headers["Content-Type"] = "application/json"
+        return case
+
+Note: A partial deep copy of the ``Case`` object is passed to each ``add_case`` hook. ``Case.endpoint.app`` and ``Case.endpoint.schema`` are shallow copied, so changes to
```

Comment:
I believe that with `Case.endpoint.app` it is adding a reference to the object rather than making a copy, but I am not completely sure. What do you think?

Reply:
That's a good point. In general, a shallow copy is the duplication of references to mutable objects. For `Case.endpoint.app`, it is the duplication of a reference to one mutable object, so it may be both. I updated the description to use "reference" for clarity.